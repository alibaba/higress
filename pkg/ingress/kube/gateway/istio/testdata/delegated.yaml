apiVersion: gateway.networking.k8s.io/v1alpha2
kind: GatewayClass
metadata:
  name: higress
spec:
  controllerName: higress.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
metadata:
  name: gateway
  namespace: higress-system
spec:
  addresses:
  - value: higress-gateway
    type: Hostname
  gatewayClassName: higress
  listeners:
  - name: apple
    hostname: "apple.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: Selector
        selector:
          matchLabels:
            kubernetes.io/metadata.name: apple
  - name: banana
    hostname: "banana.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: Selector
        selector:
          matchLabels:
            kubernetes.io/metadata.name: banana
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: http
  namespace: apple
spec:
  parentRefs:
  - name: gateway
    namespace: higress-system
  rules:
  - backendRefs:
    - name: httpbin-apple
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: http
  namespace: banana
spec:
  parentRefs:
  - name: gateway
    namespace: higress-system
  rules:
  - backendRefs:
    - name: httpbin-banana
      port: 80
---