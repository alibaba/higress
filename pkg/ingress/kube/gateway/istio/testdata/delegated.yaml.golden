apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: higress-gateway.higress-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/apple.higress-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-apple
  namespace: higress-system
spec:
  servers:
  - hosts:
    - apple/apple.example
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: higress-gateway.higress-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/banana.higress-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-banana
  namespace: higress-system
spec:
  servers:
  - hosts:
    - banana/banana.example
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/http.apple
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: http-0-istio-autogenerated-k8s-gateway
  namespace: apple
spec:
  gateways:
  - higress-system/gateway-istio-autogenerated-k8s-gateway-apple
  hosts:
  - '*'
  http:
  - name: apple/http
    route:
    - destination:
        host: httpbin-apple.apple.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/http.banana
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: http-0-istio-autogenerated-k8s-gateway
  namespace: banana
spec:
  gateways:
  - higress-system/gateway-istio-autogenerated-k8s-gateway-banana
  hosts:
  - '*'
  http:
  - name: banana/http
    route:
    - destination:
        host: httpbin-banana.banana.svc.domain.suffix
        port:
          number: 80
---
